File: multi-llm-vision.py
import requests
import sys
import json

# Supported model Ids: gemini | openai | claude

def post_image_with_params(image_path, token, text_prompt, mode):
   url = "<Request url from team APEX:E3>"


   # Prepare headers
   headers = {
       "Authorization": token,
   }


   # Prepare payload
   payload = {
       "prompt": text_prompt,
       "mode": mode,
   }


   # Open the image file
   try:
       with open(image_path, "rb") as image_file:
           files = {
               "file": image_file,
   }


           # Make the POST request
           response = requests.post(url, data=payload, headers=headers, files=files)


           # Check for success
           if response.status_code == 200:
               print("Response:", response.json())
           else:
               print(f"Failed with status code {response.status_code}: {response.text}")


   except FileNotFoundError:
       print(f"Error: The file '{image_path}' was not found.")
   except Exception as e:
       print(f"An error occurred: {e}")


if __name__ == "__main__":
  
   print(sys.argv[1])
   print(sys.argv[2])
   print(sys.argv[3])

   with open("./test-token.json", "r") as f:
    token_data = json.load(f)
    token = token_data["key"]
    if len(sys.argv) < 3:
        print("Usage: python script.py <image_path> <text_prompt> <model>")
    else:
        image_path = sys.argv[1]
        text_prompt = sys.argv[2]
        model = sys.argv[3]
        token = token
        post_image_with_params(image_path, token, text_prompt, model)

File: multi-llm-vision.js
const fs = require('fs');
const axios = require('axios');
const FormData = require('form-data');

// Supported model Ids: gemini | openai | claude

function postImageWithParams(imagePath, token, textPrompt, mode) {
  const url = "<Request url from team APEX:E3>";

  // Prepare payload
  const form = new FormData();
  form.append("prompt", textPrompt);
  form.append("mode", mode);

  try {
    const imageStream = fs.createReadStream(imagePath);
    form.append("file", imageStream);

    // Make the POST request
    axios.post(url, form, {
      headers: {
        ...form.getHeaders(),
        Authorization: token
      }
    })
    .then(response => {
      if (response.status === 200) {
        console.log("Response:", response.data);
      } else {
        console.log(`Failed with status code ${response.status}:`, response.data);
      }
    })
    .catch(error => {
      if (error.response) {
        console.error(`Request failed with status ${error.response.status}:`, error.response.data);
      } else {
        console.error("Error:", error.message);
      }
    });

  } catch (err) {
    console.error(`Error: Could not open image file '${imagePath}'.`);
  }
}

// === CLI handling ===

if (process.argv.length < 5) {
  console.log("Usage: node multi-llm-vision.js <image_path> <text_prompt> <model>");
  process.exit(1);
}

const imagePath = process.argv[2];
const textPrompt = process.argv[3];
const model = process.argv[4];

// Load token from file
fs.readFile('./test-token.json', 'utf8', (err, data) => {
  if (err) {
    console.error("Error reading token file:", err);
    return;
  }

  try {
    const tokenData = JSON.parse(data);
    const token = tokenData.key;

    postImageWithParams(imagePath, token, textPrompt, model);
  } catch (parseErr) {
    console.error("Error parsing token file:", parseErr);
  }
});
